src/Umbraco.Web/Templates/TemplateUtilities.cs

```diff
@@ -183,5 +186,57 @@ public static string ResolveMediaFromTextString(string text)
         internal static string RemoveMediaUrlsFromTextString(string text)
             // see comment in ResolveMediaFromTextString for group reference
             => ResolveImgPattern.Replace(text, "$1$3$4$5");
+
+        internal static string FindAndPersistPastedTempImages(string html, int mediaParentFolder, int userId, IMediaService mediaService, IContentTypeBaseServiceProvider contentTypeBaseServiceProvider)
```

Comment:
I think this method is unnecessarily hard to read, due to very short variable names
`f` => `file`
`fs` => `fileStream`
`absTmpImgPath` => `absoluteTemporaryImagePath`
....